src/upgrade.py

```diff
@@ -143,12 +144,17 @@ def _on_postgresql_pebble_ready(self, event: WorkloadEvent) -> None:
                 "upgrade failed. Check logs for rollback instruction"
             )
 
-    def _on_upgrade_changed(self, _) -> None:
+    def _on_upgrade_changed(self, event) -> None:
         """Update the Patroni nosync tag in the unit if needed."""
         if not self.peer_relation or not self.charm._patroni.member_started:
             return
 
         self.charm.update_config()
+        container = self.charm.unit.get_container("postgresql")
+        if not container.can_connect():
+            event.defer()
+            return
+        container.send_signal(SIGHUP, "postgresql")
```

Comment:
Secret will be generated during the Juju leader upgrade, so we should reload the config for units that upgraded earlier to pick it up.

Reply:
It sounds hackish, but no better proposal right now.

Reply:
This should be equivalent to calling the `/reload` endpoint and it shouldn't restart the DB again. We can't use the REST API or `patronictl` since creds will be in flux during the upgrade.